src/item.cpp

```diff
@@ -4024,16 +4024,31 @@ std::string item::display_name( unsigned int quantity ) const
         max_amount = to_joule( type->battery->max_capacity );
     }
 
+    std::string ammotext;
+    if( ( ( is_gun() && ammo_required() ) || is_magazine() ) && get_option<bool>( "AMMO_IN_NAMES" ) ) {
+        if( ammo_current() != "null" ) {
+            ammotext = item( ammo_current() ).ammo_type()->name();
```

Comment:
See implementation of `item::ammo_type()`. Something like this should work:

```C++
ammotext = find_type( ammo_current() )->ammo->type->name()
           ^ gets an const itype*
                                      ^ accesses itype::ammo (islot_ammo or similar)
                                            ^ get the ammunition type id from the islot_ammo
```

Reply:
This works perfectly.